product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.1-Using-switch-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.esb.scenario.test.common.http.HttpConstants;
+
+
+/**
+ * This test class test routing messages based on the Xpath with valid, Invalid, Null or empty cases
+ */
+public class RoutingBasedOnXpathTest extends ScenarioTestBase {
+
+    public static final Log log = LogFactory.getLog(RoutingBasedOnXpathTest.class);
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+
+    }
+
+    @Test(description = "5.1.1.1.1")
+    public void routeMessagesBasedOnXpathWithValidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_1_Routing_messages_based_on_content_of_message_test");
+
+        String request ="<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n" +
+                "   <m:StockName>IBM</m:StockName>\n" +
+                "</m:GetStockPrice>";
+
+        String expectedResponse = "<m:GetStockPriceResponse xmlns:m=\"http://www.example.org/stock\">\n" +
+                "    <m:Price>34.5</m:Price>\n" +
+                "</m:GetStockPriceResponse>";
+
+        HTTPUtils.invokePoxEndpointAndAssert(url, request, HttpConstants.MEDIA_TYPE_TEXT_XML, header, expectedResponse,
+                200, "Switch messages based on given Xpath with valid case name");
+    }
+
+    @Test(description = "5.1.1.1.2")
+    public void routeMessagesBasedOnXpathWithInvalidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_1_1_2_Routing_messages_based_on_Invalid_xpath_test");
+
+        String request ="<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n" +
```

Comment:
We are using the same request over and over again, right? If this is a basic xml shall we define a constant at ScenarioConstants so that it can be used across test cases?